crates/optimism/rpc/src/eth/mod.rs

```diff
@@ -273,6 +278,85 @@ impl<N: OpNodeCore> fmt::Debug for OpEthApi<N> {
     }
 }
 
+#[async_trait::async_trait]
+impl<N> L2EthApiExtServer for OpEthApi<N>
+where
+    N: OpNodeCore + 'static,
+    N::Provider: BlockReaderIdExt + StateProviderFactory,
+{
```

Comment:
we need this as a standalone struct because we want to conditionally enable this via a cli arg

e.g.: 

https://github.com/paradigmxyz/reth/blob/54c4646971f78b3c54f487931be8bb1e270bb50b/crates/optimism/rpc/src/miner.rs#L25-L25